website/docs/sdk-reference/ruby.md

```diff
@@ -417,24 +531,69 @@ Evaluates and returns the values of all feature flags and settings. Passing a [U
 | `user`     | Optional, _User Object_. Essential when using Targeting. [Read more about Targeting.](advanced/targeting.md) |
 
 ```ruby
-configcat_client = ConfigCat.create_client("#YOUR-SDK-KEY#")
-setting_values = configcat_client.get_all_values(ConfigCat::User.new("#UNIQUE-USER-IDENTIFIER#"))  # Optional User Object
+client = ConfigCat.get('#YOUR-SDK-KEY#')
+setting_values = client.get_all_values(ConfigCat::User.new('#UNIQUE-USER-IDENTIFIER#'))  # Optional User Object
+```
+
+## `get_all_value_details`
+
+Evaluates and returns the detailed values of all feature flags and settings. Passing a [User Object](#user-object) is optional.
+
+```ruby
+client = ConfigCat.get('#YOUR-SDK-KEY#')
+all_value_details = client.get_all_value_details(ConfigCat::User.new('#UNIQUE-USER-IDENTIFIER#'))  # Optional User Object
+```
+
+## Custom cache
+
+You have the option to inject your custom cache implementation into the client. All you have to do is to inherit from the ConfigCache abstract class:
+
+```ruby
+class InMemoryConfigCache < ConfigCat::ConfigCache
+    attr_reader :value
+    def initialize
+        @value = {}
+    end
+  
+    def get(key)
+        # you should return the cached value
+        return @value.fetch(key, nil)
+    end
+  
+    def set(key, value)
+        # you should cache the new value
+        @value[key] = value
+    end
+end
+```
+
+Then use your custom cache implementation:
+
+```ruby
+client = ConfigCat.get('#YOUR-SDK-KEY#',
+      ConfigCat::ConfigCatOptions.new(
+        config_cache: InMemoryConfigCache.new
+    )
+)
 ```
 
 ## Force refresh
 
-Any time you want to refresh the cached configuration with the latest one, you can call the `force_refresh()` method of the library, which initiates a new download and updates the local cache.
+Any time you want to refresh the cached configuration with the latest one, you can call the `force_refresh` method of the library, which initiates a new download and updates the local cache.
```

Comment:
Call the force refresh method on the client to download the latest config.json and update the cache.

Please fix everywhere in the docs.